contracts/lbp/LBPWrapper.sol

```diff
@@ -100,37 +83,45 @@ contract LBPWrapper {
             _swapEnabledOnStart
         );
 
-        // to handle stack overflow
-        {
-            ILBP(lbp).updateWeightsGradually(_startTime, _endTime, _endWeights);
+        ILBP(lbp).updateWeightsGradually(_startTime, _endTime, _endWeights);
+
+        return lbp;
+    }
+
+    /**
+     * @dev                          approve tokens for the vault and join pool to provide liquidity
+     * @param _tokens                array of tokens sorted for the LBP
+     * @param _amounts               amount of tokens to add to the pool to provide liquidity
+     * @param _fromInternalBalance   fund tokens from the internal user balance
+     * @param _userData              userData specifies the type of join
+     */
+    function joinPool(
+        IERC20[] memory _tokens,
+        uint256[] memory _amounts,
+        bool _fromInternalBalance,
+        bytes memory _userData
+    ) public onlyOwner {
+        require(!isPoolFunded, "LBPWrapper: pool has already been joined");
+
+        isPoolFunded = true;
+
+        address vault = address(LBPFactory.getVault());
+        for (uint8 i; i < _tokens.length; i++) {
+            IERC20(_tokens[i]).approve(vault, _amounts[i]);
         }
 
         IVault.JoinPoolRequest memory request = IVault.JoinPoolRequest({
             maxAmountsIn: _amounts,
             userData: _userData,
-            fromInternalBalance: false,
+            fromInternalBalance: _fromInternalBalance,
```

Comment:
For my own understanding: Setting `fromInternalBalance` to true basically reduces gas costs for future users that use the pool for swapping. Is that correct?

Reply:
yes, it will save gas cost. But all the users might not want to use internal balance.